### PR TITLE
Add deleteOtherAccountKeys operation to client

### PR DESCRIPTION
**Pull request by AlecRosenbaum**
The primary change here is adding `delete_other_account_keys`, which performs the `deleteOtherAccountKeys` mutation. Also included in this change are:
* slightly refactor the entity converter
* add `DeleteOtherAccountKeysPayload` entity
* add `delete_other_account_keys` client test
* group client tests into classes
* drop the `MaybeOmitted` alias in favor of the explicit Union (it wasn't
  used enough to warrant the indirection)

I have tested the mutation (by copy/pasting the body) locally, which worked as expected. I haven't done an integration test, though setting that up here is on my TODO.

### CODE REVIEW

File: authalligator_client/entities.py [hunk 1]
```diff
@@ -165,6 +165,27 @@ class Account(BaseAAEntity):
     )  # type: Optional[datetime.datetime]
 
 
+@attr.attrs(frozen=True)
+class DeleteOtherAccountKeysPayload(BaseAAEntity):
```
Comment (thomasst) [inline authalligator_client/entities.py:169]: This should eventually be a common base class for the deletion operations.

We could also just have an empty class -- if that works with attrs. There's not much point with doing all the rewriting of `_`, which useless anyways.

File: authalligator_client/client.py [hunk 2]
```diff
@@ -177,3 +182,60 @@ def query_account(
...
+                be one of the ``ProviderType`` enum values)
+            username: the AuthAlligator-provided username for the account (this
+                will likely _not_ be the human-legible email/username)
+            account_key: the AuthAlligator-specific secret key that proves we
+                have access to the specific account
+
+        Returns:
+            Not much of anything, just a placeholder boolean. Raises an
+            exception for account errors.
+        """
```
Comment (thomasst) [inline authalligator_client/client.py:208]: I'd probably do something like this:
```
Returns:
    DeleteOtherAccountKeysPayload on success.

Raises:
    AccountError in case of an account error.
```

**Review by thomasst — APPROVED** [review]

File: authalligator_client/entities.py (hunk 1 above)
Comment (AlecRosenbaum) [reply authalligator_client/entities.py:169]: I'll test it, but I'm not sure that you can initialize an empty attrs class (particularly with the entity converter). I know that mypy complained about the empty class edge case, but I ignored it since the base is never initialized in practice.

Comment (AlecRosenbaum) [reply authalligator_client/entities.py:169]: Looks like it works, so I've dropped the field.
